code/game/objects/items/weapons/tanks/tanks.dm

```diff
@@ -169,17 +169,17 @@
 				return
 			if(user.internal == src)
 				user.internal = null
-				user.internals.icon_state = "internal0"
-				usr << "<span class='notice'>You close the tank release valve.</span>"
+				user << "<span class='notice'>You close the tank release valve.</span>"
+				user.update_internals_hud_icon(0)
 				. = TRUE
 			else
 				if(user.wear_mask && (user.wear_mask.flags & MASKINTERNALS))
 					user.internal = src
-					user.internals.icon_state = "internal1"
-					usr << "<span class='notice'>You open \the [src] valve.</span>"
+					user << "<span class='notice'>You open \the [src] valve.</span>"
+					user.update_internals_hud_icon(1)
```

Comment:
there's now a proper proc to update the hud icon for internals: update_internals_hud_icon()